scheduler/scheduler.go

```diff
@@ -149,9 +150,20 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (*DependencyG
 			depGraph[d.Parent].RequiredBy, depGraph[d.Child])
 	}
 
-	return &depGraph, nil
+	cycles := DetectCycles(depGraph)
```

Comment:
Yeah, that was also my impression too, but I thought that cycle detection is too low-level operation for main func.